pkg/versionservice/versionservice.go

```diff
@@ -0,0 +1,138 @@
+package versionservice
+
+import (
+	"context"
+	"fmt"
+	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+	"github.com/percona/percona-server-mysql-operator/pkg/platform"
+	vsclient "github.com/percona/percona-server-mysql-operator/pkg/versionservice/client"
+	"github.com/percona/percona-server-mysql-operator/pkg/versionservice/client/models"
+	"github.com/percona/percona-server-mysql-operator/pkg/versionservice/client/version_service"
+	"github.com/pkg/errors"
+	"net/http"
+	"net/url"
+	"time"
+
+	_ "github.com/Percona-Lab/percona-version-service/api"
+)
+
+const (
+	productName     = "ps-operator"
+	defaultEndpoint = "https://check.percona.com"
+)
+
+func GetExactVersion(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL, serverVersion *platform.ServerVersion) (DepVersion, error) {
+	endpoint := cr.Spec.UpgradeOptions.VersionServiceEndpoint
+	if endpoint == "" {
+		endpoint = defaultEndpoint
+	}
+
+	requestURL, err := url.Parse(endpoint)
+	if err != nil {
+		return DepVersion{}, errors.Wrap(err, "url parse")
+	}
+
+	client := vsclient.NewHTTPClientWithConfig(nil, &vsclient.TransportConfig{
```

Comment:
Could we maybe initialize this client one at startup and pass it rather than initializing it every 5 seconds by calling it from the controller?

Reply:
@inelpandzic The problem is that we should update the endpoint for the client. I don't think that the initialization of the client is so heavy operation that we should increase the complexity of the code. If the endpoint has been static, it would be ok.